src/cript/nodes/core.py

```diff
@@ -72,3 +73,73 @@ def json(self):
         from cript.nodes.util import NodeEncoder
 
         return json.dumps(self, cls=NodeEncoder)
+
+    def find_children(self, search_attr:dict, recursion_depth:int=-1) -> List:
+        """
+        Finds all the children in a given tree of nodes (specified by its root),
+        that match the criteria of search_attr.
+        If a node is present multiple times in the graph, it is only once in the search results.
+
+        recursion_dept: Max depth of recursion into the tree. Helpful if circles are expected. -1 specifies no limit
```

Comment:
I recommend removing the word `recursion` from docstrings because it could give a future developer an idea that this is using recursion when it is using iteration.

Reply:
done, replaced it by search depth